magento/magento2
# Solve fatal error on credit memo mail

Also insert the correct item for invoice mail, to calculate the appropriate subtotal on row item.

This is an addition of PR https://github.com/magento/magento2/pull/22265 where the subtotal for creditmemo should be fixed. Now this same strategy is chosen for the invoice mail and the fatal error caused by this previous PR is solved.

By checking the type instance of `$item` I tried to keep it as strict as possible while keeping code the change as simple as possible.

### Fixed Issues (if relevant)
1. magento/magento2#25617: Fatal error on creditmemo mail grouped product

### Manual testing scenarios (*)
1. Add grouped product to cart, make sure to have for one item 3 products.
2. Place order. 
3. In the backend create invoice for this order, **nb** make sure to invoice only 2 of the 3 products (update qty)
4. Send invoice mail, see that the subtotal is correct
5. In the backend create credit memo for this same order, **nb** make sure to only credit 1 product (update qty)
6. Send credit memo mail, see that the subtotal is correct.


### Contribution checklist (*)
